docker/docker.ts

```diff
@@ -545,15 +568,22 @@ async function runCommandThatCanFail(
         });
 
         p.on("error", err => {
-            // If we actually full on received some sort of error in the process, also try to dump
-            // any stderr messages in case they might be useful.
-            logStdErrMessages(/*code: */ 1);
-            reject(err);
+            stdErrChunks.push(new Buffer(err.message));
+            finish(/*code: */ 1);
         });
 
         p.on("close", code => {
-            logStdErrMessages(code);
+            finish(code);
```

Comment:
I don't think it's safe to call finish in both `error` and `close`.  I believe both will frequently get called?

And it looks like we now never `reject` - is that really intentional?